sdk/keyvault/azure-security-keyvault-keys/src/key_type.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "azure/keyvault/keys/key_type.hpp"
+
+#include <stdexcept>
+
+using namespace Azure::Security::KeyVault::Keys;
+
+KeyTypeEnum Details::KeyTypeFromString(std::string const& name)
+{
+  if (name == EcValue)
+  {
+    return KeyTypeEnum::Ec;
+  }
+  if (name == EcHsmValue)
+  {
+    return KeyTypeEnum::EcHsm;
+  }
+  if (name == OctValue)
+  {
+    return KeyTypeEnum::Oct;
+  }
+  if (name == OctHsmValue)
+  {
+    return KeyTypeEnum::OctHsm;
+  }
+  if (name == RsaValue)
+  {
+    return KeyTypeEnum::Rsa;
+  }
+  if (name == RsaHsmValue)
+  {
+    return KeyTypeEnum::RsaHsm;
+  }
+  throw std::runtime_error("cannot convert " + name + " to key type (kty)");
+}
+
+std::string Details::KeyTypeToString(KeyTypeEnum kty)
+{
+  if (kty == KeyTypeEnum::Ec)
```

Comment:
Or maybe write this function as switch/case, and initialize a static map for the reverse function by feeding string values into this function. Or build an inverse map, one from another.